src/test/java/com/example/rets_api/repository/BasicPropertyWithOneEntityRepositoryJPATest.java

```diff
@@ -0,0 +1,240 @@
+package com.example.rets_api.repository;
+
+
+import com.example.rets_api.entity.PropertyEntity;
+import com.example.rets_api.resource.Enums;
+import com.example.rets_api.utils.EntityUtilsTest;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static com.example.rets_api.utils.EntityUtilsTest.*;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+public class BasicPropertyWithOneEntityRepositoryJPATest{
+
+    BaseTests baseTests = new BaseTests();
+    @Autowired
+    PropertyRepositoryJPA propertyRepository;
+
+    @Test
+    public void should_store_a_PropertyAndFinancialData(){
+        //Creating
+        PropertyEntity propertyToSave = EntityUtilsTest.createPropertyEntityWithBasicFields();
+        propertyToSave.setFinancialData(createFinancialDataEntity());
+        PropertyEntity propertyEntityToCompare = EntityUtilsTest.createPropertyEntityWithAllNestedFields();
```

Comment:
You can create inside EntityUtilsTest one method for each nested object, for instance, createPropertyEntityWithBasicFieldsAndFinancialData(), createPropertyEntityWithBasicFieldsAndViewData()...

Reply:
done